framework/src/interface/packet.rs

```diff
@@ -422,7 +425,24 @@ impl<T: EndOffset, M: Sized + Send> Packet<T, M> {
                     ptr::copy_nonoverlapping(fin_dst, move_loc, to_move);
                 }
                 ptr::copy_nonoverlapping(hdr, fin_dst as *mut T2, 1);
-                Ok(())
+                Ok(header_size as isize)
+            } else {
+                Err(ErrorKind::FailedToInsertHeader.into())
+            }
+        }
+    }
+
+    #[inline]
+    pub fn insert_header_fn<T2: EndOffset<PreviousHeader = T>>(
+        &mut self,
+        header: &T2,
+        on_insert: &Fn(&mut T, isize),
```

Comment:
oh yay, didn't get to this in the rust tutorial 👍

Reply:
the deref magic on getting the header there was fun haha.